src/test/java/com/iexec/sms/tee/EnableIfTeeProviderTests.java

```diff
@@ -0,0 +1,183 @@
+package com.iexec.sms.tee;
+
+import com.iexec.sms.tee.config.GramineInternalServicesConfiguration;
+import com.iexec.sms.tee.config.SconeInternalServicesConfiguration;
+import com.iexec.sms.tee.config.TeeInternalServicesConfiguration;
+import com.iexec.sms.tee.session.generic.TeeSessionHandler;
+import com.iexec.sms.tee.session.gramine.GramineSessionHandlerService;
+import com.iexec.sms.tee.session.gramine.GramineSessionMakerService;
+import com.iexec.sms.tee.session.gramine.sps.SpsConfiguration;
+import com.iexec.sms.tee.session.scone.SconeSessionHandlerService;
+import com.iexec.sms.tee.session.scone.SconeSessionMakerService;
+import com.iexec.sms.tee.session.scone.SconeSessionSecurityConfig;
+import com.iexec.sms.tee.session.scone.cas.CasClient;
+import com.iexec.sms.tee.session.scone.cas.CasConfiguration;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.springframework.context.annotation.ConditionContext;
+import org.springframework.context.annotation.Conditional;
+import org.springframework.core.env.Environment;
+import org.springframework.core.type.AnnotationMetadata;
+
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.when;
+
+class EnableIfTeeProviderTests {
+    @Conditional(EnableIfTeeProvider.class)
+    @EnableIfTeeProviderDefinition(providers = {})
+    static class NoProvidersSet {
+
```

Comment:
Is there a reason for this empty line ?

Reply:
No reason, except that we sometimes start our classes with an empty line. We should define some good practice and stick to it.

Reply:
I don't know if they are good practice. I often put an empty line when beginning a new class, I find it more readable.
For empty classes/methods, I often go for a one liner.
here it would become
`static class NoProviderSet {}`

Reply:
Right for the empty classes. See https://github.com/iExecBlockchainComputing/iexec-sms/pull/119/commits/40876587f75a9032d7707dcf9cfb28406f0968f7 :)